frontend/src/metabase/components/FieldValuesWidget.jsx

```diff
@@ -541,3 +436,201 @@ const OptionsMessage = ({ message }) => (
 OptionsMessage.propTypes = optionsMessagePropTypes;
 
 export default connect(mapStateToProps, mapDispatchToProps)(FieldValuesWidget);
+
+// if [dashboard] parameter ID is specified use the fancy new Chain Filter API endpoints to fetch parameter values.
+// Otherwise (e.g. for Cards) fall back to the old field/:id/values endpoint
+function usesChainFilterEndpoints(dashboard) {
+  return dashboard?.id;
+}
+
+function showRemapping(fields) {
+  return fields.length === 1;
+}
+
+function shouldList(fields, disableSearch) {
```

Comment:
I know it's a transitional refactoring, but just in case, watch out for Boolean trap: https://ariya.io/2011/08/hall-of-api-shame-boolean-trap.